app/controllers/comment_controller.rb

```diff
@@ -321,4 +317,8 @@ def check_permission_or_redirect!(comment, target)
                         action: target.show_action, id: target.id)
     false
   end
+
+  def safe_model_from_param_type
+    Comment.all_types.find { |m| m.name == params[:type] }
+  end
```

Comment:
Just an idea -- why not move this into Comment, then Query will have access to it, too.

Reply:
Good idea. Maybe I should rename it `safe_constantize`. :laughing: